Classes/GarbageCollector.php

```diff
@@ -215,6 +219,33 @@ protected function deleteIndexDocuments($table, $uid)
     }
 
     /**
+     * Delete all index documents that are related to a TYPO3 "page"
+     *
+     * @param $uid
+     */
+    protected function deleteRecordsOfPage($uid)
+    {
+        /** @var \ApacheSolrForTypo3\Solr\ConnectionManager $connectionManager */
+        $connectionManager = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\ConnectionManager');
+        $indexQueue = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\IndexQueue\\Queue');
+        $site = Site::getSiteByPageId($uid);
+
+        // a site can have multiple connections (cores / languages)
+        $solrConnections = $connectionManager->getConnectionsBySite($site);
+
+        /** @var  $solr */
+        foreach ($solrConnections as $solr) {
+            $response = $solr->search('pid:' . intval($uid), 0, 10000);
```

Comment:
Ah true, didn't think about folder pages. However, if you have a page you should be able to get its site configuration and the indexing configuration to find what types to look for. I was first thinking about simply looking at the Index Queue, but that doesn't have a page ID. Adding it wouldn't make things easier either since then you'd have to update it whenever a record is moved.

Thinking about it further it might actually be possible; since we get notified of any changes to a record we could update the Index Queue's entry's pageID whenever we update the entry. 

Then you'd have an easy way to find records on a page and the documents that belong to it. 

Also, how about using the reference index, could that work?